File: apps/agent/src/ingest/embedder.ts
import { embed } from "../lib/vertex.js";

const BATCH = 5;

export async function embedBatch(texts: string[]): Promise<number[][]> {
  const out: number[][] = [];
  for (let i = 0; i < texts.length; i += BATCH) {
    const slice = texts.slice(i, i + BATCH);
    const vectors = await embed(slice);
    out.push(...vectors);
  }
  return out;
}

File: apps/agent-py/app/ingest/embedder.py
"""Batched embedding — port of apps/agent/src/ingest/embedder.ts."""
from __future__ import annotations

from app.llm.genai_client import embed

BATCH = 5


async def embed_batch(texts: list[str]) -> list[list[float]]:
    out: list[list[float]] = []
    for i in range(0, len(texts), BATCH):
        out.extend(await embed(texts[i:i + BATCH]))
    return out
